Automattic/wp-calypso
# Signup: new flow for Business and Premium signup

This fixes #2734 by adding a new signup flow which removes the free option in the plans step, and adds the free trial option for both Premium and Business plans.

This also includes some refactoring to the way we handle Jetpack plans, cc @richardmuscat 

#### Note
This now provides a way for users to sign up for a free trial without being in the test. Perhaps we should also stop the test?

#### Testing instructions - New signup
- Open `/start/plan` and go through the new plan selection flow.
- [x] Assert that you see two placeholders while the plans load
- [x] Assert that you do see free trials on the plan selection and plan comparison steps
- [x] Assert that you don't see the free plan on the plan selection and plan comparison steps

#### Testing instructions - Default signup
- Open `/start` and go through the main signup flow.
- [x] Assert that you see three placeholders while the plans load
- [x] Assert that you don't see free trials on the plan selection and plan comparison steps
- [x] Assert that you do see the free plan on the plan selection and plan comparison steps

#### Testing instructions - Free trial signup - offered
- Open `/start/free-trial` and go through the free trial flow.
- Add yourself to the test: `localStorage.setItem('ABTests','{"freeTrials_20160120":"offered"}')`
- [x] Assert that you see three placeholders while the plans load
- [x] Assert that you do see free trials on the plan selection and plan comparison steps
- [x] Assert that you do see the free plan on the plan selection and plan comparison steps

#### Testing instructions - Jetpack
- Open the `/plans` page for a Jetpack site
- [x] Assert that you see two placeholders while the plans load
- [x] Assert that you don't see the free plan - only Jetpack Business and Jetpack Premium, on both the plan selection and plan comparison screen

#### Testing instructions - WPCOM plans
- Open the `/plans` page for a WPCOM site
- [x] Assert that you see three placeholders while the plans load
- [x] Assert that you see all three WPCOM plans on the plan selection and plan comparison screens.
- [x] Assert that you are not offered a free trial

#### Additional tests
- [x] If you are in the free trials test you will see free trials on the /start/free-trial flow and in /plans/:site and will be redirect to the free-trial signup flow when visiting /start.
- [x] If you are not in the free trials test you will not see free trials on the /start/free-trial flow, on /plans/:site  or in the default signup flow.
- [x] If you are not in the free trials test then the only place you can see free trials is on /start/plan

#### Reviews
